hadoop-common-project/hadoop-common/pom.xml

```diff
@@ -381,73 +381,43 @@
     </resources>
     <plugins>
       <plugin>
-        <groupId>org.apache.hadoop</groupId>
-        <artifactId>hadoop-maven-plugins</artifactId>
+        <groupId>org.xolstice.maven.plugins</groupId>
+        <artifactId>protobuf-maven-plugin</artifactId>
         <executions>
           <execution>
-            <id>version-info</id>
-            <phase>generate-resources</phase>
-            <goals>
-              <goal>version-info</goal>
-            </goals>
-            <configuration>
-              <source>
-                <directory>${basedir}/src/main</directory>
-                <includes>
-                  <include>java/**/*.java</include>
-                  <include>proto/**/*.proto</include>
-                </includes>
-              </source>
-            </configuration>
+            <id>src-compile-protoc</id>
+            <configuration><skip>false</skip></configuration>
           </execution>
           <execution>
-            <id>compile-protoc</id>
+            <id>compile-test-protoc</id>
+            <phase>generate-test-resources</phase>
             <goals>
-              <goal>protoc</goal>
+              <goal>test-compile</goal>
             </goals>
             <configuration>
-              <protocVersion>${protobuf.version}</protocVersion>
-              <protocCommand>${protoc.path}</protocCommand>
-              <imports>
-                <param>${basedir}/src/main/proto</param>
-              </imports>
-              <source>
-                <directory>${basedir}/src/main/proto</directory>
-                <includes>
-                  <include>HAServiceProtocol.proto</include>
-                  <include>IpcConnectionContext.proto</include>
-                  <include>ProtocolInfo.proto</include>
-                  <include>RpcHeader.proto</include>
-                  <include>ZKFCProtocol.proto</include>
-                  <include>ProtobufRpcEngine.proto</include>
-                  <include>Security.proto</include>
-                  <include>GetUserMappingsProtocol.proto</include>
-                  <include>TraceAdmin.proto</include>
-                  <include>RefreshAuthorizationPolicyProtocol.proto</include>
-                  <include>RefreshUserMappingsProtocol.proto</include>
-                  <include>RefreshCallQueueProtocol.proto</include>
-                  <include>GenericRefreshProtocol.proto</include>
-                  <include>FSProtos.proto</include>
-                </includes>
-              </source>
+              <protoTestSourceRoot>${basedir}/src/test/proto</protoTestSourceRoot>
```

Comment:
Is protoTestSourceRoot a valid configuration for this plugin?

Reply:
Yes. Found here https://www.xolstice.org/protobuf-maven-plugin/test-compile-mojo.html#protoTestSourceRoot

Reply:
Yes. Found here https://www.xolstice.org/protobuf-maven-plugin/test-compile-mojo.html#protoTestSourceRoot